src/backend/TrafficCourts/Workflow.Service/Consumers/VirusScanDocumentConsumer.cs

```diff
@@ -0,0 +1,94 @@
+﻿using MassTransit;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json.Linq;
+using System.Net;
+using TrafficCourts.Common.OpenAPIs.VirusScan.V1;
+using TrafficCourts.Coms.Client;
+using TrafficCourts.Messaging.MessageContracts;
+using TrafficCourts.Workflow.Service.Services;
+using ApiException = TrafficCourts.Common.OpenAPIs.VirusScan.V1.ApiException;
+
+namespace TrafficCourts.Workflow.Service.Consumers
+{
+    /// <summary>
+    /// Consumer for VirusScanDocument message.
+    /// </summary>
+    public class VirusScanDocumentConsumer : IConsumer<VirusScanDocument>
+    {
+        private readonly IVirusScanService _virusScanService;
+        private readonly IComsService _comsService;
+        private readonly ILogger<VirusScanDocumentConsumer> _logger;
+
+        public VirusScanDocumentConsumer(IVirusScanService virusScanService, IComsService comsService, ILogger<VirusScanDocumentConsumer> logger)
+        {
+            _virusScanService = virusScanService ?? throw new ArgumentNullException(nameof(virusScanService));
+            _comsService = comsService ?? throw new ArgumentNullException(nameof(comsService));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public async Task Consume(ConsumeContext<VirusScanDocument> context)
+        {
+            using var scope = _logger.BeginConsumeScope(context);
+
+            _logger.LogDebug("Consuming VirusScanDocument message");
+            var cancellationToken = context.CancellationToken;
+
+            Guid documentId = context.Message.DocumentId;
+
+            try
+            {
+                // Get the file to scan from COMS
+                Coms.Client.File file = await _comsService.GetFileAsync(documentId, cancellationToken);
+
+                var stream = file.Data;
+                // Reset position to the beginning of the stream
+                stream.Position = 0;
+
+                // Virus scan the file and get scan result
+                _logger.LogDebug("Sending the file: {fileId} for virus scan", file.Id);
+                ScanResponse scanResult = await _virusScanService.ScanDocumentAsync(stream, cancellationToken);
+
+                if (VirusScanStatus.NotInfected.Equals(scanResult.Status))
+                {
+                    _logger.LogDebug("No viruses detected as a result of the scan");
+                    // Add a "clean" metadata tag to the document if no viruses detected
+                    file.Metadata.Add("virus-scan-status", "clean");
```

Comment:
change these `Add` calls to just use the set via indexer,` file.Metadata["virus-scan-status"] = "clean";` but also remove other properties if they exist like "virus-name".  this helps if we re-run the virus scan and these properties already exist.

Reply:
Done.